cdap-test/src/main/java/co/cask/cdap/test/TestManager.java

```diff
@@ -573,4 +575,19 @@ void deployDatasetModule(DatasetModuleId datasetModuleId, Class<? extends Datase
    * @param namespaceId the namespace from which to remove all apps
    */
   void deleteAllApplications(NamespaceId namespaceId) throws Exception;
+
+  /**
+   * Add a new schedule to a existing application
+   *
+   * @param scheduleId the ID of the schedule to add
+   * @param scheduleDetail the {@link ScheduleDetail} describing the new schedule.
+   */
+  void addSchedule(ScheduleId scheduleId, ScheduleDetail scheduleDetail) throws Exception;
```

Comment:
why not accept `ApplicationId` instead of `ScheduleId`?

Reply:
The `ScheduleClient` takes a `ScheduleId` in both add and delete schedule. I think `ScheduleId` is more useful than `ApplicationId` since we will anyway construct a ScheduleId but we may not use the appId right?